src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -1231,6 +1231,11 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 						if (connectionMngInfo.serverInfo) {
 							connection.options.isCloud = connectionMngInfo.serverInfo.isCloud;
 						}
+						else if (connection.userName.length === 0) {
```

Comment:
1. check for sqllogin type only
2. make sure username is configured to be required.